NewsFeed/view_models/CCBRNewsViewModel.m

```diff
@@ -19,21 +25,30 @@ - (instancetype)initWithDataSource:(id<CCBRArticleDataSource>)dataSource {
         self.dataSource = dataSource;
         
         __weak CCBRNewsViewModel *weakSelf = self;
-        self.dataSource.nextArticlesCallback = ^(NSUInteger startIndex, NSUInteger endIndex) {
+        self.dataSource.nextArticlesCallback = ^(NSUInteger startIndex, NSUInteger endIndex, NSError *error) {
+            self.error = error;
             if (weakSelf.updateCallback) {
-                weakSelf.updateCallback();
+                weakSelf.updateCallback(startIndex, endIndex);
             }
```

Comment:
These lines of code will cause the `collection view` to update whether there's any error or not. This is not optimal and can cause unexpected consequences.

For clarity, we should separate `error flow` and `success flow`.